### DISCUSSION TITLE
Add future date ranges for date interval filter template

### DISCUSSION THREAD

**Discussion by phil-maynard** [microsoft-search/pnp-modern-search#3685] [Ideas]
It would be really useful if the built-in 'date interval' filter template accommodated future ranges e.g. Next week, Next month, Next quarter, Next year.

**Current issue**
The current date interval template only presents the end use past date range options - e.g. 'Past 245 hours', 'From past week to past month' - even if the values returned in the results are for future dates. For search pages that contain items with future dates that need to be used as filters, this limits the options available to end users.

**Desired solution**
**_Option 1:_** Update the 'date interval' template to include standard future intervals
**_Option 2:_** Provide a mechanism for designers to define pre-set filter options in a custom layout

**Case**
I'm building a search page for an Events calendar where the salient information is for upcoming events. Although the date range template works, the stakeholder would like easier common date intervals to be available in the filter.

An alternative would be to have enhanced functionality to specify a custom template using fixed filters with dynamic ranges e.g.:
data-name="Next week" 
data-value="range({Today},{Today+7}"

**Example of a desired refiner**
![image](https://github.com/microsoft-search/pnp-modern-search/assets/29703291/0fd7e867-7592-4fe4-ac97-9ca16be22399)

**Comment by kasperbolarsen** [comment]
Hi 👋 
I agree, that would be an awesome custom refiner layout.
If you are able to, please add it as a PR to the custom layouts repository here https://github.com/microsoft-search/pnp-modern-search-layouts

**Reply by phil-maynard** [reply]
Hi @kasperbolarsen I've raised this idea on the bases that I'm not aware of a way to add additional filter options via a custom layout. Am I correct in thinking that custom layouts can only amend returned filters as in the [Nation flags refiner example](https://github.com/microsoft-search/pnp-modern-search-layouts/blob/main/Refiner/Nation%20flags%20in%20refiner/Nation%20Flags%20in%20the%20refiner.md) or is there a way to manually add additional filter options?

**Comment by kasperbolarsen** [comment]
Hey, I have had a look at the source code and as far as I can tell, \pnp-modern-search\search-parts\src\components\filters\FilterDateIntervalComponent.tsx has an allOptions enum and none of these are in the future. I *guess* this can be changed without to many problems, but then again, I ain't a Dev and can't do it.

**Comment by kanade-saket** [comment]
@phil-maynard  , Thanks for this idea . I came across similar requirement for custom range filter and contemplating some ideas. 
 
If i am getting this correct , there are 2 places where we need change to handle this , 1st in the query where refiners are set like below screenshot  from >> pnp-modern-search\search-parts\src\dataSources\SharePointSearchDataSource.ts 
2nd Part is filter template >> which i think we can handle through extensibility  by adding additional filter web components (checkbox template works ok , but option text is not user friendly)

**Possible Solution could be -**  adding a advanced filter condition where Users can directly set the refiner (with support to some placeholders for date) , this can be also used for numeric range filter and may be also for graph range filter , 
if not through UI , at least through Edit raw properties

@wobba  , Highly appreciate if you can provide some guidance , Thanks in advance .

![image](https://github.com/microsoft-search/pnp-modern-search/assets/114055549/9128b502-ed6f-456e-865d-89e5bb41b05b)
